OptimizelySDKCore/OptimizelySDKCoreTests/OptimizelyTest.m

```diff
@@ -382,6 +382,84 @@ - (void)testIsFeatureEnabledWithFeatureFlagEnabledAndUserIsBeingExperimented {
+    
+    XCTAssertFalse([self.optimizely isFeatureEnabled:featureFlagKey userId:kUserId attributes:nil], @"should return false for disabled featureFlag");
+    
+    OCMVerify([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]);
+    [decisionServiceMock stopMocking];
+}
+
+// Should return true if the feature experiment variation’s `featureEnabled` property is true
+- (void)testIsFeatureEnabledWithVariationsFeatureEnabledtrue {
+    NSString *featureFlagKey = @"booleanFeature";
+    OPTLYFeatureFlag *featureFlag = [self.optimizely.config getFeatureFlagForKey:featureFlagKey];
+    OPTLYExperiment *experiment = [self.optimizely.config getExperimentForId:featureFlag.experimentIds[1]];
+    OPTLYVariation *variation = experiment.variations[1];
+    OPTLYFeatureDecision *decision = [[OPTLYFeatureDecision alloc] initWithExperiment:experiment variation:variation source:DecisionSourceExperiment];
+    
+    id decisionServiceMock = OCMPartialMock(self.optimizely.decisionService);
+    
+    OCMStub([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]).andReturn(decision);
+    
+    XCTAssertTrue([self.optimizely isFeatureEnabled:featureFlagKey userId:kUserId attributes:nil], @"should return true for enabled featureFlag");
+    
+    OCMVerify([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]);
+    
+    [decisionServiceMock stopMocking];
+}
+
+// Should return true if the user is bucketed into rollout experiment’s variation
+// and variation's featureEnabled is also true
+- (void)testIsFeatureEnabledWithVariationsFeatureEnabledtrueForRollout {
+    NSString *featureFlagKey = @"booleanSingleVariableFeature";
+    OPTLYRollout *rollout = [self.optimizely.config getRolloutForId:@"166660"];
+    OPTLYExperiment *experiment = rollout.experiments[0];
+    OPTLYVariation *variation = experiment.variations[0];
+    OPTLYFeatureFlag *featureFlag = [self.optimizely.config getFeatureFlagForKey:featureFlagKey];
+    OPTLYFeatureDecision *decision = [[OPTLYFeatureDecision alloc] initWithExperiment:experiment variation:variation source:DecisionSourceRollout];
+    
+    id decisionServiceMock = OCMPartialMock(self.optimizely.decisionService);
+    
+    OCMStub([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]).andReturn(decision);
+    
+    XCTAssertTrue([self.optimizely isFeatureEnabled:featureFlagKey userId:kUserId attributes:nil], @"should return true for enabled featureFlag");
+    
+    OCMVerify([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]);
+    [decisionServiceMock stopMocking];
+}
+
+// Should return false if the user is bucketed into rollout experiment’s variation
+// but variation's featureEnabled is false
+- (void)testIsFeatureEnabledWithVariationsFeatureEnabledForRollout {
```

Comment:
Comment on the code comment very nearly like the preceding comment on a code comment.  In this case:

Comment on the code comment.  Our design doc requests:

Test that `isFeatureEnabled` returns false if the user is NOT bucketed into the rollout experiment’s variation

I've also read in the design doc that "featureEnabled only appears in variations under experiments
which are feature experiments".  I infer from the design doc that rollout experiment variations will
not contain the key "featureEnabled".  It is the decision service response from `isFeatureEnabled`
which will be false.

Reply:
CAN UNIT TESTS SHOULD CHECK FOR PRESENCE/ABSENCE OF "featureEnabled"?

If the design doc's statement "featureEnabled only appears in variations under experiments
which are feature experiments" does mean "rollout experiment" variations don't contain
"featureEnabled", then I think it is best if we can arrange our unit tests to verify this claim
for both "feature experiment"s and "rollout experiment"s.  Specifically:
* "feature experiment" variations should contain "featureEnabled"
* "rollout experiment" variations should not contain "featureEnabled"